Migrate to AGP 3.0

I already tried couple of times and failed miserably 😔

hi guys, i have DEX build an error with gradle 3.0

stacktrace

```
Caused by: com.android.dex.DexException: Multiple dex files define Lcom/pushtorefresh/storio2/R;
        at com.android.dx.merge.DexMerger.readSortableTypes(DexMerger.java:661)
        at com.android.dx.merge.DexMerger.getSortedTypes(DexMerger.java:616)
        at com.android.dx.merge.DexMerger.mergeClassDefs(DexMerger.java:598)
        at com.android.dx.merge.DexMerger.mergeDexes(DexMerger.java:171)
        at com.android.dx.merge.DexMerger.merge(DexMerger.java:198)
        at com.android.builder.dexing.DexArchiveMergerCallable.call(DexArchiveMergerCallable.java:61)
```

--- a/build.gradle
+++ b/build.gradle
@@ -6,11 +6,12 @@ buildscript {
     ext.kotlin_version = '1.1.51'
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
         // Android Gradle plugin
-        classpath 'com.android.tools.build:gradle:2.3.3'
+        classpath 'com.android.tools.build:gradle:3.0.1'
 
         // Jacoco Test Coverage Plugin
         // (addition to standard Jacoco plugin, allows us fail the build if coverage is not enough)
@@ -33,6 +34,7 @@ buildscript {
 allprojects {
     repositories {
         jcenter()
+        google()
     }
 
     // Workaround for preventing Gradle from stealing focus from other apps
@@ -43,21 +45,17 @@ allprojects {
 }
 
 ext {
-    buildToolsVersion = '25.0.2'
-    compileSdkVersion = 25
+    compileSdkVersion = 27
     minSdkVersion = 14
 }
 
-def supportLibsVersion = '25.3.1'
+def supportLibsVersion = '27.0.2'
 def daggerVersion = '2.4'
 
 ext.libraries = [
         // Core libraries
         supportAnnotations                       : 'com.android.support:support-annotations:' + supportLibsVersion,
 
-        // TODO remove RxJava1 and Interop once RxJava2 migration will be complete.
-        rxJava                                   : 'io.reactivex:rxjava:1.2.1',
-        rxJavaInterop                            : 'com.github.akarnokd:rxjava2-interop:0.10.9',
         rxJava2                                  : 'io.reactivex.rxjava2:rxjava:2.1.6',
         rxAndroid2                               : 'io.reactivex.rxjava2:rxandroid:2.0.1',
         kotlinStdLib                             : "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version",
@@ -83,7 +81,7 @@ ext.libraries = [
         equalsVerifier                           : 'nl.jqno.equalsverifier:equalsverifier:1.7.2',
         privateConstructorChecker                : 'com.pushtorefresh.java-private-constructor-checker:checker:1.1.0',
         guava                                    : 'com.google.guava:guava:18.0',
-        robolectric                              : 'org.robolectric:robolectric:3.3.2',
+        robolectric                              : 'org.robolectric:robolectric:3.5.1',
         googleTestingCompile                     : 'com.google.testing.compile:compile-testing:0.11',
 
         dagger                                   : 'com.google.dagger:dagger:' + daggerVersion,
@@ -96,8 +94,8 @@ ext.libraries = [
         appCompat                                : 'com.android.support:appcompat-v7:' + supportLibsVersion,
         recyclerView                             : 'com.android.support:recyclerview-v7:' + supportLibsVersion,
         timber                                   : 'com.jakewharton.timber:timber:3.0.1',
-        leakCanary                               : 'com.squareup.leakcanary:leakcanary-android:1.3.1',
-        leakCanaryNoOp                           : 'com.squareup.leakcanary:leakcanary-android-no-op:1.3.1',
+        leakCanary                               : 'com.squareup.leakcanary:leakcanary-android:1.5.4',
+        leakCanaryNoOp                           : 'com.squareup.leakcanary:leakcanary-android-no-op:1.5.4',
 
         autoService                              : 'com.google.auto.service:auto-service:1.0-rc2',
         javaPoet                                 : 'com.squareup:javapoet:1.8.0',
@@ -107,8 +105,6 @@ ext.libraries = [
 
         autoParcel                               : 'com.github.frankiesardo:auto-parcel:0.3',
         autoParcelProcessor                      : 'com.github.frankiesardo:auto-parcel-processor:0.3'
-
-
 ]
 
 def getRuntimeJar() {

--- a/gradle/publish-android-lib.gradle
+++ b/gradle/publish-android-lib.gradle
@@ -94,13 +94,14 @@ afterEvaluate { project ->
     }
 
     task androidJavadocs(type: Javadoc) {
-        source = android.sourceSets.main.java.srcDirs
-
-        classpath += project.files(android.getBootClasspath())
-        classpath += configurations.compileJavadoc
-        classpath += configurations.compile + configurations.provided
-
-        failOnError false // Sorry, but we can not make it work with dependencies between modules
+        android.libraryVariants.all { variant ->
+            task("${variant.name}Javadoc", type: Javadoc) {
+                description "Generates Javadoc for $variant.name."
+                source = variant.javaCompile.source
+                classpath = files(variant.javaCompile.classpath.files,
+                        project.android.getBootClasspath())
+            }
+        }
 
         exclude '**/BuildConfig.java'
         exclude '**/R.java'

--- a/storio-basic-sample-app/build.gradle
+++ b/storio-basic-sample-app/build.gradle
@@ -2,7 +2,6 @@ apply plugin: 'com.android.application'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         applicationId 'com.pushtorefresh.storio.basic_sample'
@@ -14,6 +13,11 @@ android {
         testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
     }
 
+    buildTypes {
+        debug { matchingFallbacks = ['release'] }
+        release {}
+    }
+
     lintOptions {
         disable 'GoogleAppIndexingWarning'
     }
@@ -21,13 +25,13 @@ android {
 
 // libraries declared in root build.gradle
 dependencies {
-    compile libraries.appCompat
-    compile libraries.recyclerView
+    implementation libraries.appCompat
+    implementation libraries.recyclerView
 
-    compile libraries.storIOSQLite
+    implementation libraries.storIOSQLite
 
-    compile libraries.storIOSQLiteAnnotations
+    implementation libraries.storIOSQLiteAnnotations
     annotationProcessor libraries.storIOSQLiteAnnotationsProcessor
 
-    testCompile  libraries.junit
+    testImplementation libraries.junit
 }

--- a/storio-common-annotations-processor/build.gradle
+++ b/storio-common-annotations-processor/build.gradle
@@ -6,12 +6,11 @@ dependencies {
     compile libraries.javaPoet
     compile libraries.kotlinStdLib
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.mockitoKotlin
-    testCompile libraries.equalsVerifier
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.mockitoKotlin
+    testImplementation libraries.equalsVerifier
 }
-
 
 sourceSets {
     main.java.srcDirs += 'src/main/kotlin'

--- a/storio-common/build.gradle
+++ b/storio-common/build.gradle
@@ -3,7 +3,6 @@ apply plugin: 'checkstyle'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         versionName VERSION_NAME
@@ -23,15 +22,17 @@ android {
 }
 
 dependencies {
-    compile libraries.supportAnnotations
-    provided libraries.rxJava2
+    api libraries.supportAnnotations
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.mockitoCore
-    testCompile libraries.equalsVerifier
-    testCompile libraries.privateConstructorChecker
-    testCompile libraries.storIOTestCommon
+    compileOnly libraries.rxJava2
+    testImplementation libraries.rxJava2
+
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.mockitoCore
+    testImplementation libraries.equalsVerifier
+    testImplementation libraries.privateConstructorChecker
+    testImplementation libraries.storIOTestCommon
 }
 
 apply from: '../gradle/checkstyle.gradle'

--- a/storio-content-resolver-annotations-processor-test/build.gradle
+++ b/storio-content-resolver-annotations-processor-test/build.gradle
@@ -3,7 +3,6 @@ apply from: '../build.workaround-missing-resource.gradle'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         minSdkVersion rootProject.ext.minSdkVersion
@@ -11,9 +10,10 @@ android {
 }
 
 dependencies {
-    testCompile libraries.storIOContentResolver
-    testCompile libraries.storIOContentResolverAnnotationsProcessor
-    testCompile libraries.googleTestingCompile
-    testCompile libraries.runtimeJar
-    testCompile libraries.toolsJar
+    testImplementation      libraries.storIOContentResolver
+    testImplementation      libraries.storIOContentResolverAnnotationsProcessor
+    testAnnotationProcessor libraries.storIOContentResolverAnnotationsProcessor
+    testImplementation      libraries.googleTestingCompile
+    testImplementation      libraries.runtimeJar
+    testImplementation      libraries.toolsJar
 }

--- a/storio-content-resolver/build.gradle
+++ b/storio-content-resolver/build.gradle
@@ -2,7 +2,6 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         versionName VERSION_NAME
@@ -22,17 +21,19 @@ android {
 }
 
 dependencies {
-    compile  libraries.storIOCommon
-    provided libraries.rxJava2
+    api libraries.storIOCommon
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.assertJAndroid
-    testCompile libraries.mockitoCore
-    testCompile libraries.equalsVerifier
-    testCompile libraries.privateConstructorChecker
-    testCompile libraries.storIOTestCommon
-    testCompile libraries.robolectric
+    compileOnly libraries.rxJava2
+    testImplementation libraries.rxJava2
+
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.assertJAndroid
+    testImplementation libraries.mockitoCore
+    testImplementation libraries.equalsVerifier
+    testImplementation libraries.privateConstructorChecker
+    testImplementation libraries.storIOTestCommon
+    testImplementation libraries.robolectric
 }
 
 apply from: '../gradle/checkstyle.gradle'

--- a/storio-content-resolver/src/test/java/com/pushtorefresh/storio3/contentresolver/integration/IntegrationTest.java
+++ b/storio-content-resolver/src/test/java/com/pushtorefresh/storio3/contentresolver/integration/IntegrationTest.java
@@ -11,7 +11,7 @@ import com.pushtorefresh.storio3.contentresolver.impl.DefaultStorIOContentResolv
 import org.junit.Before;
 import org.robolectric.Robolectric;
 import org.robolectric.RuntimeEnvironment;
-import org.robolectric.util.ContentProviderController;
+import org.robolectric.android.controller.ContentProviderController;
 
 public abstract class IntegrationTest {
 

--- a/storio-sample-app/build.gradle
+++ b/storio-sample-app/build.gradle
@@ -3,7 +3,6 @@ apply plugin: 'com.squareup.sqldelight'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         applicationId 'com.pushtorefresh.storio.sample'
@@ -27,6 +26,7 @@ android {
     buildTypes {
         debug {
             signingConfig signingConfigs.sample
+            matchingFallbacks = ['release']
         }
 
         release {
@@ -46,37 +46,49 @@ android {
     }
 }
 
-dependencies {
-    compile libraries.storIOSQLite
-    compile libraries.storIOContentResolver
+// Ensure the no-op dependency is always used in JVM tests.
+configurations.all { config ->
+    if (config.name.contains('UnitTest')) {
+        config.resolutionStrategy.eachDependency { details ->
+            if (details.requested.group == 'com.squareup.leakcanary' && details.requested.name == 'leakcanary-android') {
+                details.useTarget(group: details.requested.group, name: 'leakcanary-android-no-op', version: details.requested.version)
+            }
+        }
+    }
+}
 
-    compile libraries.storIOSQLiteAnnotations
+dependencies {
+    implementation libraries.storIOSQLite
+    implementation libraries.storIOContentResolver
+
+    implementation libraries.storIOSQLiteAnnotations
     annotationProcessor libraries.storIOSQLiteAnnotationsProcessor
 
-    compile libraries.storIOContentResolverAnnotations
+    implementation libraries.storIOContentResolverAnnotations
     annotationProcessor libraries.storIOContentResolverAnnotationsProcessor
 
-    compile libraries.dagger
+    implementation libraries.dagger
     annotationProcessor libraries.daggerCompiler
 
-    compile libraries.javaxInject
-    compile libraries.javaxAnnotationApi
+    implementation libraries.javaxInject
+    implementation libraries.javaxAnnotationApi
 
-    compile libraries.rxJava2
-    compile libraries.rxAndroid2
-    compile libraries.appCompat
-    compile libraries.butterKnife
-    compile libraries.recyclerView
-    compile libraries.timber
+    implementation libraries.rxJava2
+    implementation libraries.rxAndroid2
+    implementation libraries.appCompat
+    implementation libraries.recyclerView
+    implementation libraries.timber
+    implementation libraries.butterKnife
+    annotationProcessor libraries.butterKnife
 
-    provided libraries.autoParcel
+    compileOnly libraries.autoParcel
     annotationProcessor libraries.autoParcelProcessor
 
-    debugCompile libraries.leakCanary
-    releaseCompile libraries.leakCanaryNoOp
+    debugImplementation libraries.leakCanary
+    releaseImplementation libraries.leakCanaryNoOp
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.robolectric
-    testCompile libraries.leakCanaryNoOp
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.robolectric
+    testImplementation libraries.leakCanaryNoOp
 }

--- a/storio-sample-app/src/main/java/com/pushtorefresh/storio3/sample/ui/activity/BaseActivity.java
+++ b/storio-sample-app/src/main/java/com/pushtorefresh/storio3/sample/ui/activity/BaseActivity.java
@@ -3,7 +3,7 @@ package com.pushtorefresh.storio3.sample.ui.activity;
 import android.support.annotation.LayoutRes;
 import android.support.annotation.NonNull;
 import android.support.annotation.Nullable;
-import android.support.v7.app.ActionBarActivity;
+import android.support.v7.app.AppCompatActivity;
 import android.support.v7.widget.Toolbar;
 import android.view.MenuItem;
 
@@ -12,7 +12,7 @@ import com.pushtorefresh.storio3.sample.R;
 import io.reactivex.disposables.CompositeDisposable;
 import io.reactivex.disposables.Disposable;
 
-public abstract class BaseActivity extends ActionBarActivity {
+public abstract class BaseActivity extends AppCompatActivity {
 
     @Nullable
     private Toolbar toolbar;

--- a/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/SampleRobolectricTestRunner.java
+++ b/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/SampleRobolectricTestRunner.java
@@ -1,0 +1,24 @@
+package com.pushtorefresh.storio3.sample;
+
+import android.support.annotation.NonNull;
+
+import org.junit.runners.model.InitializationError;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.annotation.Config;
+
+import java.lang.reflect.Method;
+
+public class SampleRobolectricTestRunner extends RobolectricTestRunner {
+
+    public SampleRobolectricTestRunner(Class<?> testClass) throws InitializationError {
+        super(testClass);
+    }
+
+    @Override
+    @NonNull
+    public Config getConfig(@NonNull Method method) {
+        return new Config.Builder(super.getConfig(method))
+                .setApplication(SampleApp.class)
+                .build();
+    }
+}

--- a/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/db/DbOpenHelperTest.java
+++ b/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/db/DbOpenHelperTest.java
@@ -3,16 +3,16 @@ package com.pushtorefresh.storio3.sample.db;
 import android.database.sqlite.SQLiteDatabase;
 
 import com.pushtorefresh.storio3.contentresolver.BuildConfig;
+import com.pushtorefresh.storio3.sample.SampleRobolectricTestRunner;
 
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.robolectric.RobolectricTestRunner;
 import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.Config;
 
 import static org.assertj.core.api.Java6Assertions.assertThat;
 
-@RunWith(RobolectricTestRunner.class)
+@RunWith(SampleRobolectricTestRunner.class)
 @Config(constants = BuildConfig.class, sdk = 21)
 public final class DbOpenHelperTest {
 

--- a/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/sample_code/RelationsTest.java
+++ b/storio-sample-app/src/test/java/com/pushtorefresh/storio3/sample/sample_code/RelationsTest.java
@@ -2,13 +2,13 @@ package com.pushtorefresh.storio3.sample.sample_code;
 
 import com.pushtorefresh.storio3.contentresolver.BuildConfig;
 import com.pushtorefresh.storio3.sample.SampleApp;
+import com.pushtorefresh.storio3.sample.SampleRobolectricTestRunner;
 import com.pushtorefresh.storio3.sample.db.entities.Tweet;
 import com.pushtorefresh.storio3.sample.db.entities.TweetWithUser;
 import com.pushtorefresh.storio3.sample.db.entities.User;
 
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.robolectric.RobolectricTestRunner;
 import org.robolectric.RuntimeEnvironment;
 import org.robolectric.annotation.Config;
 
@@ -17,7 +17,7 @@ import java.util.List;
 import static java.util.Arrays.asList;
 import static org.assertj.core.api.Java6Assertions.assertThat;
 
-@RunWith(RobolectricTestRunner.class)
+@RunWith(SampleRobolectricTestRunner.class)
 @Config(constants = BuildConfig.class, sdk = 21)
 public class RelationsTest {
 

--- a/storio-sqlite-annotations-processor-test/build.gradle
+++ b/storio-sqlite-annotations-processor-test/build.gradle
@@ -3,7 +3,6 @@ apply from: '../build.workaround-missing-resource.gradle'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         minSdkVersion rootProject.ext.minSdkVersion
@@ -11,9 +10,9 @@ android {
 }
 
 dependencies {
-    testCompile libraries.storIOSQLite
-    testCompile libraries.storIOSQLiteAnnotationsProcessor
-    testCompile libraries.googleTestingCompile
-    testCompile libraries.runtimeJar
-    testCompile libraries.toolsJar
+    testImplementation libraries.storIOSQLite
+    testImplementation libraries.storIOSQLiteAnnotationsProcessor
+    testImplementation libraries.googleTestingCompile
+    testImplementation libraries.runtimeJar
+    testImplementation libraries.toolsJar
 }

--- a/storio-sqlite/build.gradle
+++ b/storio-sqlite/build.gradle
@@ -2,7 +2,6 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         versionName VERSION_NAME
@@ -18,19 +17,22 @@ android {
 }
 
 dependencies {
-    compile  libraries.storIOCommon
-    provided libraries.rxJava2
+    api                 libraries.storIOCommon
 
-    testCompile  libraries.junit
-    testCompile  libraries.assertJ
-    testCompile  libraries.mockitoCore
-    testCompile  libraries.equalsVerifier
-    testCompile  libraries.storIOTestCommon
-    testCompile  libraries.privateConstructorChecker
-    testCompile  libraries.guava
-    testCompile  libraries.robolectric
-    testCompile  libraries.autoParcel
-    testProvided libraries.autoParcelProcessor
+    compileOnly         libraries.rxJava2
+    testImplementation  libraries.rxJava2
+
+    testImplementation  libraries.junit
+    testImplementation  libraries.assertJ
+    testImplementation  libraries.mockitoCore
+    testImplementation  libraries.equalsVerifier
+    testImplementation  libraries.storIOTestCommon
+    testImplementation  libraries.privateConstructorChecker
+    testImplementation  libraries.guava
+    testImplementation  libraries.robolectric
+    testImplementation  libraries.autoParcel
+    testImplementation  libraries.autoParcelProcessor
+    testAnnotationProcessor libraries.autoParcelProcessor
 }
 
 apply from: '../gradle/checkstyle.gradle'

--- a/storio-sqlite/src/test/java/com/pushtorefresh/storio3/sqlite/impl/DefaultStorIOSQLiteTest.java
+++ b/storio-sqlite/src/test/java/com/pushtorefresh/storio3/sqlite/impl/DefaultStorIOSQLiteTest.java
@@ -4,6 +4,7 @@ import android.content.ContentValues;
 import android.database.sqlite.SQLiteDatabase;
 import android.database.sqlite.SQLiteOpenHelper;
 import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
 
 import com.pushtorefresh.storio3.TypeMappingFinder;
 import com.pushtorefresh.storio3.internal.ChangesBus;
@@ -31,8 +32,6 @@ import java.lang.reflect.Modifier;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.Set;
-
-import javax.annotation.Nullable;
 
 import io.reactivex.Scheduler;
 import io.reactivex.subscribers.TestSubscriber;

--- a/storio-test-common/build.gradle
+++ b/storio-test-common/build.gradle
@@ -2,7 +2,6 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         minSdkVersion rootProject.ext.minSdkVersion
@@ -20,15 +19,18 @@ android {
 }
 
 dependencies {
-    provided libraries.supportAnnotations
-    provided libraries.rxJava2
+    compileOnly libraries.supportAnnotations
 
-    compile libraries.junit
-    compile libraries.mockitoCore
+    implementation libraries.junit
+    implementation libraries.mockitoCore
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.privateConstructorChecker
+    compileOnly libraries.rxJava2
+    testImplementation libraries.rxJava2
+
+    testImplementation libraries.supportAnnotations
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.privateConstructorChecker
 }
 
 apply from: '../gradle/checkstyle.gradle'

--- a/storio-test-without-rxjava/build.gradle
+++ b/storio-test-without-rxjava/build.gradle
@@ -2,7 +2,6 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion rootProject.ext.compileSdkVersion
-    buildToolsVersion rootProject.ext.buildToolsVersion
 
     defaultConfig {
         minSdkVersion rootProject.ext.minSdkVersion
@@ -20,15 +19,15 @@ android {
 }
 
 dependencies {
-    compile libraries.storIOSQLite
-    compile libraries.storIOContentResolver
+    implementation libraries.storIOSQLite
+    implementation libraries.storIOContentResolver
 
-    provided libraries.supportAnnotations
+    compileOnly libraries.supportAnnotations
 
-    testCompile libraries.junit
-    testCompile libraries.assertJ
-    testCompile libraries.mockitoCore
-    testCompile libraries.storIOTestCommon
+    testImplementation libraries.junit
+    testImplementation libraries.assertJ
+    testImplementation libraries.mockitoCore
+    testImplementation libraries.storIOTestCommon
 }
 
 apply from: '../gradle/checkstyle.gradle'

